### PR TITLE
updating flowtype lib defs

### PR DESCRIPTION
**Pull request by alexreardon**
I will need some help with this one. 

@thejameskyle can I grab some of your time to help with this one? I feel like I am entering dark waters

### CODE REVIEW

**Comment by alexreardon** [comment]
Any flow ninjas out there keen to lend a hand on this one?

**Comment by thejameskyle** [comment]
Happy to help, just grab me at some point

**Comment by alexreardon** [comment]
Getting close!

File: src/state/hook-middleware.js [hunk 1]
```diff
@@ -104,10 +112,10 @@ const getFireHooks = (hooks: Hooks) => memoizeOne((current: State, previous: Sta
 
 export default (hooks: Hooks) => {
   const fireHooks = getFireHooks(hooks);
-  return (store: Store) => (next: (Action) => mixed) => (action: Action): mixed => {
+  return (store: Store) => (next: Dispatch) => (action: Action): Action => {
```
Comment (alexreardon) [inline src/state/hook-middleware.js]: This is saying that `Store` does not have the right type...

**Pull request merged**